pkg/e2e/infra/infra.go

```diff
@@ -195,9 +195,22 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
 		err = waitForClusterSizeToBeHealthy(client, initialClusterSize)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		workerNode, err := getWorkerNode(client)
+		g.By("getting worker node")
+		var workerNode *corev1.Node
+		o.Eventually(func() bool {
+			workerNodes, err := e2e.GetWorkerNodes(client)
+			if err != nil {
+				return false
+			}
+			if len(workerNodes) == 0 {
+				return false
+			}
+			workerNode = &workerNodes[0]
+			return true
+		}, time.Minute, time.Second).ShouldNot(o.BeTrue())
```

Comment:
Should not be true? I would expect the opposite.

Reply:
nice catch, done